src/Components/Components/src/ComponentBase.cs

```diff
@@ -52,8 +52,9 @@ protected virtual void BuildRenderTree(RenderTreeBuilder builder)
         {
```

Comment:
Technically we could make `BuildRenderTree` into an `abstract` method now if we wanted. I don't personally feel strongly about doing so, since the built-in empty implementation is a legal default that you might even want to use in some scenarios (e.g., for components that don't produce any visible output, but exist only to cause side-effects on inclusion/disposal).

Reply:
I think virtual is fine as-is.